### PR TITLE
[Refactor][meta-format-change]Change backup/restore job to json format

### PR DESCRIPTION
**Pull request by gengjun-git**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 101 / 121 (83.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[1429]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|[343, 344, 345, 353, 354, 355]|
|:large_blue_circle:|com/starrocks/backup/Repository.java|7|15|46.67%|[719, 720, 722, 730, 731, 732, 733, 737]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|2|50.00%|[1425]|
|:large_blue_circle:|com/starrocks/backup/RepositoryMgr.java|3|5|60.00%|[172, 173]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|22|24|91.67%|[702, 703]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|67|67|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/backup/Repository.java [hunk 1]
```diff
@@ -702,22 +710,33 @@ public void readFields(DataInput in) throws IOException {
...
+        try {
+            st = storage.checkPathExist(listPath);
+        } catch (Exception e) {
+            LOG.warn("check path exist fail");
+            prefixRepo = "__starrocks_repository_";
+            return;
+        }
+
+        if (st.ok()) {
+            prefixRepo = "__palo_repository_";
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/backup/Repository.java:737]: can we remove this completely and if not, why?

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java [hunk 2]
```diff
@@ -671,6 +677,38 @@ public long loadBackupHandler(DataInputStream dis, long checksum, GlobalStateMgr
         return checksum;
     }
 
+    public void saveBackupHandlerV2(DataOutputStream dos) throws IOException, SRMetaBlockException {
+        SRMetaBlockWriter writer = new SRMetaBlockWriter(dos,
+                BackupHandler.class.getName(), 2 + dbIdToBackupOrRestoreJob.size());
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java:682]: using fixed string name?

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/backup/Repository.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/backup/Repository.java:737]: We should keep this logic, the change from __palo_repository_ to __starrocks_repository_ is started from version 3.0

File: fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/backup/BackupHandler.java:682]: The following PR will change theses name by once.

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 101 / 121 (83.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[1439]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|6|00.00%|[343, 344, 345, 353, 354, 355]|
|:large_blue_circle:|com/starrocks/backup/Repository.java|7|15|46.67%|[719, 720, 722, 730, 731, 732, 733, 737]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|2|50.00%|[1425]|
|:large_blue_circle:|com/starrocks/backup/RepositoryMgr.java|3|5|60.00%|[172, 173]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|22|24|91.67%|[702, 703]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|67|67|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**